### PR TITLE
MINOR: use method references instead of anonymus classes in Errors

### PR DESCRIPTION
**Pull request by viktorsomogyi**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/common/protocol/Errors.java [hunk 1]
```diff
@@ -109,531 +109,98 @@
...
+    DELEGATION_TOKEN_EXPIRED(66, "Delegation Token is expired.", DelegationTokenExpiredException::new),
+    INVALID_PRINCIPAL_TYPE(67, "Supplied principalType is not supported", InvalidPrincipalTypeException::new),
+    NON_EMPTY_GROUP(68, "The group is not empty", GroupNotEmptyException::new),
+    GROUP_ID_NOT_FOUND(69, "The group id does not exist", GroupIdNotFoundException::new),
+    FETCH_SESSION_ID_NOT_FOUND(70, "The fetch session ID was not found", FetchSessionIdNotFoundException::new),
+    INVALID_FETCH_SESSION_EPOCH(71, "The fetch session epoch is invalid", InvalidFetchSessionEpochException::new),
+    LISTENER_NOT_FOUND(72, "There is no listener on the leader broker that matches the listener on which metadata request was processed", ListenerNotFoundException::new),;
 
+    @FunctionalInterface
     private interface ApiExceptionBuilder {
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/common/protocol/Errors.java]: Do we need this at all or can we use one of the Java 8 functional interfaces?

Comment (viktorsomogyi) [reply clients/src/main/java/org/apache/kafka/common/protocol/Errors.java]: Hmm, thinking of it, we could use `java.util.Function<String, ApiException>`. Let me update this PR to see how it looks.

**Comment by ijuma** [comment]
Both PR builds failed. Restarting them as the results are no longer available. Also, can you please ensure the lines are not too long? They should fit within the GitHub review window.

**Comment by ijuma** [comment]
retest this please

**Comment by viktorsomogyi** [comment]
Ok, I'll try to fit the lines better.

**Review by ijuma — APPROVED** [review]
Thanks for the PR, LGTM.

**Pull request merged**